dipy/reconst/multi_voxel.py

```diff
@@ -71,13 +72,18 @@ def new_fit(self, data, mask=None, **kwargs):
                 data_to_fit[ii : ii + vox_per_chunk]
                 for ii in range(0, data_to_fit.shape[0], vox_per_chunk)
             ]
+            parallel_kwargs = {}
+            for kk in ["n_jobs", "vox_per_chunk", "engine", "verbose"]:
+                if kk in kwargs:
+                    parallel_kwargs[kk] = kwargs[kk]
             fit_array[np.where(mask)] = np.concatenate(
                 (
                     paramap(
                         _parallel_fit_worker,
                         chunks,
                         func_args=[single_voxel_with_self],
-                        **kwargs,
+                        func_kwargs=kwargs,
```

Comment:
So can you please explain, that if I would like to pass in different keywords for each chunk, how to do this?

For example, I would need something like `weights=weights[ii : ii + vox_per_chunk]` such that each parallel worker is received a different weights `keyword`. I am still not sure how to achieve this here - would kwargs be replaced with a list of dictionaries, with each dictionary specifying the keywords for that chunk? Thanks

Reply:
Yes. I think that's right: a list of dictionaries.